### DISCUSSION TITLE
How to customize igx-legend

### DISCUSSION THREAD

**Discussion by KaushikEDS** [IgniteUI/igniteui-angular#11250] [Q&A]
## Question  
I want to use igx legend in financial charts. 
Although I want to customize the design and such. For ex. Adding tooltips, Hide/Unhide from legend, Changing the design of the legend, etc.
I couldn't find it anywhere in the documentation.
I would like a way to manipulate the legend according to my various use cases.

**Comment by agoldenbaum** [comment]
Hello KaushikEDS,

In order to customize the design of the legend, you will need to provide your own legendItemTemplate for the series of the IgxFinancialChart. The legendItemTemplate property can be found on the series, and so you will need to handle the seriesAdded event of the chart. You can also modify the badge by similarly setting the legendItemBadgeTemplate property of the series.

I am attaching a sample project to demonstrate a modification of the legendItemTemplate of the series in the IgxFinancialChart.

Please let me know if you have any other questions or concerns on this matter.

Sincerely,
Andrew Goldenbaum
Software Developer – Team Lead
Infragistics

[demo.zip](https://github.com/IgniteUI/igniteui-angular/files/7884185/demo.zip)

**Reply by KaushikEDS** [reply]
@agoldenbaum 

I checked the demo.zip provided and I couldn't understand a few things.

1. What are LegendTemplates and what does resolveLegendTemplate do?
![image](https://user-images.githubusercontent.com/79090369/175586087-488430e2-c9f4-4d66-80a3-5cdbce792bf4.png)
2. What are obj and child variables? How can I use them?
![image](https://user-images.githubusercontent.com/79090369/175598094-ec4053f4-731f-4f34-b489-0b0c62b1bbd5.png)





I want to do the following things in legend - 

1. Customize the text displayed in the legend
2. Add custom elements (like the checkbox in the demo.zip)
4. Change the position and orientation of the legend in the chart
5. Customize the badge for legend.
